utils/syntax_highlighting.tsx

```diff
@@ -204,3 +80,203 @@ function getLanguageFromNameOrAlias(name: string) {
         return aliases && aliases.find((a) => a === langName);
     });
 }
+
+async function registerLanguage(languageName: string) {
+    let language: LanguageFn | undefined;
+    switch (languageName) {
```

Comment:
Suggestion for structuring this a bit differently, I don't think we can construct the import path dynamically (I feel like I've tried that before unsuccessfully), but could we convert this into a dictionary to reduce the number of lines here and get rid of all the break statements?

I think something like this should help keep the number of lines down while still being fairly readable as long as you make sure to not run the 

```typescript
const languageImports = {
    '1c': () => import('highlight.js/lib/languages/1c'),
    actionscript: () => import('highlight.js/lib/languages/actionscript'),
    // ...
};

if (!languageImports[languageName]) {
    return;
}

const language = (await langaugeImports[languageName]()).default;
```

Reply:
I used map instead of switch.